From 45a046e86437a70938f416ea85651e18dae066e4 Mon Sep 17 00:00:00 2001
From: Joseph Chen <chenjh@rock-chips.com>
Date: Mon, 23 Oct 2017 14:49:04 +0800
Subject: [PATCH] regulator: rk808: delete rk816 boost and otg switch regulator

Due to hardware issue, boost and otg switch must be conctrolled
with special timing carefully by rk816 battery driver. Otherwise
boost maybe burn off.

Change-Id: Id28208c5b9757e1ff0e57ec5d26f1ac0afb88ad5
Signed-off-by: Joseph Chen <chenjh@rock-chips.com>
---
 drivers/regulator/rk808-regulator.c | 19 -------------------
 include/linux/mfd/rk808.h           |  2 +-
 2 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/drivers/regulator/rk808-regulator.c b/drivers/regulator/rk808-regulator.c
index 6e0619903add..d827c4165e54 100644
--- a/drivers/regulator/rk808-regulator.c
+++ b/drivers/regulator/rk808-regulator.c
@@ -729,25 +729,6 @@ static const struct regulator_desc rk816_reg[] = {
 		.disable_val = BIT(7),
 		.owner = THIS_MODULE,
 	},
-	RK816_DESC(RK816_ID_BOOST, "DCDC_BOOST", "boost", 4700, 5400, 100,
-		   RK816_BOOST_ON_VESL_REG, RK818_BOOST_ON_VSEL_MASK,
-		   RK816_DCDC_EN_REG2, BIT(4) | BIT(0), BIT(4) | BIT(0),
-		   BIT(4), 0),
-	{
-		.name = "OTG_SWITCH",
-		.supply_name = "usb",
-		.of_match = of_match_ptr("OTG_SWITCH"),
-		.regulators_node = of_match_ptr("regulators"),
-		.id = RK816_ID_OTG_SWITCH,
-		.ops = &rk808_switch_ops,
-		.type = REGULATOR_VOLTAGE,
-		.n_voltages = 1,
-		.enable_reg = RK816_DCDC_EN_REG2,
-		.enable_mask = BIT(6) | BIT(2),
-		.enable_val = BIT(6) | BIT(2),
-		.disable_val = BIT(6),
-		.owner = THIS_MODULE,
-	},
 	RK816_DESC(RK816_ID_LDO1, "LDO_REG1", "vcc5", 800, 3400, 100,
 		   RK816_LDO1_ON_VSEL_REG, RK818_LDO_VSEL_MASK,
 		   RK816_LDO_EN_REG1, BIT(4) | BIT(0), BIT(4) | BIT(0),
diff --git a/include/linux/mfd/rk808.h b/include/linux/mfd/rk808.h
index 7c876c09336c..64d61fac51d6 100644
--- a/include/linux/mfd/rk808.h
+++ b/include/linux/mfd/rk808.h
@@ -549,7 +549,7 @@ enum rk805_reg {
 #define RK816_CHIP_NAME_REG			0x17
 #define RK816_CHIP_VER_REG			0x18
 #define RK816_OTP_VER_REG			0x19
-#define RK816_NUM_REGULATORS			12
+#define RK816_NUM_REGULATORS			10
 
 /*POWER ON/OFF REGISTER*/
 #define RK816_VB_MON_REG			0x21
-- 
2.35.3

